content/operators-workshop/02-CRDs.md

```diff
@@ -0,0 +1,52 @@
+# Third Party Resources (Legacy)
+
+- First mechanism for extending Kubernetes
+- Added in 1.2
+- Deprecated in 1.7
+- Removed in 1.8
+
+---
+
+# Custom Resource Definition
+
+- Added as beta in 1.7
+- Reimplemented based on lessons learned from TPRs
+- Namespace or cluster scoped
+- Better pluralization support
+- Shortname support
+
+---
+
+# Example CRD
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: postgresqlconfigs.containerconf.de
+spec:
+  group: containerconf.de
+  version: v1
+  scope: Namespaced
+  names:
+    plural: postgresqlconfigs
+    singular: postgresqlconfig 
+    kind: PostgreSQLConfig
+    shortNames: []
+```
+
+---
+
+# API endpoint
+
+- An endpoint is created for the CRD
+
+```
+/apis/containerconf.de/v1/postgresqlconfigs
+```
+
+- Similar to the standard endpoints
+
+```
+/api/v1/pods
+```
```

Comment:
I think we should show evolution of the API - core (no group) & named groups. And say CRD creates new endpoint in the named group.

The core group, often referred to as the legacy group, is at the REST path /api/v1 and uses apiVersion: v1.
Example: `/api/v1/pods`

The named groups are at REST path /apis/$GROUP_NAME/$VERSION, and use apiVersion: $GROUP_NAME/$VERSION (e.g. apiVersion: batch/v1).
Example: `/apis/apps/v1beta1/deployments`

Reply:
Good point I've updated to include this.